src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -233,13 +233,12 @@ def validate_user(namespace):
         raise CLIError("--user can only be clusterUser or clusterMonitoringUser")
 
 
-def validate_vnet_subnet_id(namespace):
-    if namespace.vnet_subnet_id is not None:
-        if namespace.vnet_subnet_id == '':
-            return
-        from msrestazure.tools import is_valid_resource_id
-        if not is_valid_resource_id(namespace.vnet_subnet_id):
-            raise CLIError("--vnet-subnet-id is not a valid Azure resource ID.")
+def validate_subnet_id(namespace):
+    if namespace.vnet_subnet_id is None or namespace.vnet_subnet_id == '':
+        return
+    from msrestazure.tools import is_valid_resource_id
+    if not is_valid_resource_id(namespace.vnet_subnet_id):
+        raise CLIError("--vnet-subnet-id is not a valid Azure resource ID.")
```

Comment:
Can you change it to `InvalidArgumentValueError` as we are deprecating `CLIError`? Details can be found in [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

Reply:
Done though it seems like changing all of them should be a seperate change. so onl changed this function.